skyplane/obj_store/hdfs_interface.py

```diff
@@ -0,0 +1,107 @@
+from functools import lru_cache
+from pyarrow import fs
+from dataclasses import dataclass
+from typing import Iterator, List, Optional
+from skyplane.exceptions import NoSuchObjectException
+from skyplane.obj_store.object_store_interface import ObjectStoreInterface, ObjectStoreObject
+import mimetypes
+
+
+@dataclass
+class HDFSFile(ObjectStoreObject):
+    def full_path(self):
+        return f"hdfs://{self.key}"
+
+
+class HDFSInterface(ObjectStoreInterface):
+    def __init__(self, path, port=8020):
+        self.hdfs_path = path
+        self.port = port
+        self.hdfs = fs.HadoopFileSystem(host=self.hdfs_path, port=self.port, user="hadoop", extra_conf={"dfs.permissions.enabled": "false"})
+
+    def path(self) -> str:
+        return self.hdfs_path
+
+    def list_objects(self, prefix="") -> Iterator[HDFSFile]:
+        fileSelector = fs.FileSelector(prefix=prefix, recursive=True)
+        response = self.hdfs.get_file_info(fileSelector)
+        for file in response:
+            yield HDFSFile(key=file.path, size=file.size, last_modified=file.mtime)
+
+    def exists(self, obj_name: str):
+        try:
+            self.get_obj_metadata(obj_name)
+            return True
+        except NoSuchObjectException:
+            return False
+        
+    def region_tag(self) -> str:
+        return ""
+
+    def bucket(self) -> str:
```

Comment:
In the previous meeting, we said that we can think of buckets as directories. However, considering that we are stateless with files, would keeping track of the entered directory (when they initially initiate the connection) be okay?